### PR TITLE
Refactor out animateFloatMergePredictiveBackAsState

### PR DESCRIPTION
**Pull request by nullArrayList**

### CODE REVIEW

**Comment by FooIbar** [comment]
Screenshot_20231109-103244_EhViewer_Debug 1
Feels like too high

**Comment by nullArrayList** [comment]
> Screenshot_20231109-103244_EhViewer_Debug 1 Feels like too high

Reverted

**Comment by FooIbar** [comment]
Breaks navigation bar padding

**Comment by nullArrayList** [comment]
> Breaks navigation bar padding

嗯？哪break了

**Comment by FooIbar** [comment]
> > Breaks navigation bar padding
> 
> 嗯？哪break了
Screenshot_20231109-104759_EhViewer_Debug 1

EditText has fixed height

**Comment by nullArrayList** [comment]
> > > Breaks navigation bar padding
> > 
> > 
> > 嗯？哪break了
> 
> Screenshot_20231109-104759_EhViewer_Debug 1
> 
> EditText has fixed height

Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 1]
```diff
@@ -258,9 +258,10 @@ fun GalleryCommentsScreen(galleryDetail: GalleryDetail, navigator: NavController
         },
     ) { paddingValues ->
         val keylineMargin = dimensionResource(id = R.dimen.keyline_margin)
+        var editTextMeasured by remember { mutableStateOf(80.dp) }
         Box(modifier = Modifier.fillMaxSize().imePadding()) {
             val additionalPadding = if (commenting) {
-                EditTextHeight
+                editTextMeasured
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:264]: Double navigation bar padding for comment list

Comment (nullArrayList) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:264]: Done

**Comment by FooIbar** [comment]
https://github.com/FooIbar/EhViewer/assets/118464521/0cad9a94-fb41-450e-ac80-247c8a7c6d47

Comment list jumps

**Comment by nullArrayList** [comment]
> screen-20231109-112329.1.mp4 
> Comment list jumps

Done

**Comment by FooIbar** [comment]
Screenshot_20231109-115557_EhViewer_Debug 1
Screenshot_20231109-115631_EhViewer_Debug 1
Too far with load more button and too close without it.

**Comment by nullArrayList** [comment]
> Screenshot_20231109-115557_EhViewer_Debug 1 Screenshot_20231109-115631_EhViewer_Debug 1 Too far with load more button and too close without it.

Done

**Comment by FooIbar** [comment]
Still jumps when there are no more comments

**Comment by nullArrayList** [comment]
> Still jumps when there are no more comments

Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 2]
```diff
@@ -390,7 +390,7 @@ fun GalleryCommentsScreen(galleryDetail: GalleryDetail, navigator: NavController
             ) {
                 Row(
                     modifier = Modifier.fillMaxWidth().navigationBarsPadding().onGloballyPositioned { coordinates ->
-                        editTextMeasured = max(with(density) { coordinates.size.height.toDp() }, 80.dp)
+                        editTextMeasured = max(with(density) { coordinates.size.height.toDp() }, 88.dp)
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:393]: Extract this out?
Others LGTM.

**Review by FooIbar — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt (hunk 2 above)
Comment (nullArrayList) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:393]: 88.dp？

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:393]: Yeah

Comment (nullArrayList) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt]: Done

**Pull request merged**